buyandsell/views.py

```diff
@@ -0,0 +1,185 @@
+
+
+    def update_limits(self):
+        for limit in Limit.objects.all():
+            if(limit.endtime != None and limit.endtime<timezone.localtime()):
+                limit.delete()
+    
+    def update_bans(self, product:Product=None):
+        if(product!=None):
+            """Get the existing reports on this product that have been accepted by the moderator
+            but have not been addressed (by initiating a ban)."""
+            reports = Report.objects.filter(product=product, addressed=False, accepted=True)
+            if(len(reports)>=REPORTS_THRES):
+                """Create a ban for the user lasting three days."""
+                endtime = timezone.localtime()+timezone.timedelta(days=3)
+                Ban.objects.create(user=product.user, endtime=endtime)
+                product.deleted=True
+                reports.update(addressed=True)
+        else:
+            """Calls the above if-block on products that have accepted but unaddressed reports."""
+            reports = Report.objects.filter(accepted=True, addressed=False)
+            products = []
+            for report in reports:
+                products.append(report.product)
+            for prod in set(products):
+                if(products.count(prod)>=REPORTS_THRES):
+                    """Products from a banned user cannot be reported. (acc. to report function.)"""
+                    endtime = timezone.localtime()+timezone.timedelta(days=3)
+                    Ban.objects.create(user=product.user, endtime=endtime)
+                    reports.update(addressed=True)
+    
+    def category_filter(self, request, queryset):
+        category = request.GET.get('category')
+        if(category!=None and len(Category.objects.filter(name=category))>0):
+            queryset = queryset.filter(category__name=category)
+        return queryset
+    def seller_filter(self,request, queryset):
+        seller = request.GET.get('seller')
+        if(seller!=None and len(UserProfile.objects.filter(ldap_id=seller))>0):
+            queryset = queryset.filter(user=UserProfile.objects.get(ldap_id=seller))
+        return queryset
+    def list(self, request):
+        ##introduce tags?
+        self.update_bans()
+        queryset = self.queryset.filter(status=True)
+        """remove products from banned users"""
+        bans = Ban.objects.all()
+        for ban in bans:
+            if(ban.endtime>timezone.localtime()):
+                """Remove products from users whose bans are still running."""
+                queryset = queryset.filter(~Q(user=ban.user))
+        #TODO: allow category to be passed here too.
+        queryset = query_search(request, 3, queryset, ['name', 'description'], 'buyandsell')
+        queryset = self.category_filter(request, queryset)
+        queryset = self.seller_filter(request, queryset)
+        # queryset = query_from_num(request, self.RESULTS_PER_PAGE, queryset)
```

Comment:
Remove this as it won't be used. If pagination is required, check out how its done in placementblog.